win_images/win-server-wsl.yml

```diff
@@ -0,0 +1,87 @@
+variables:
+  # Naming suffix for images to prevent clashes
+  IMG_SFX:
+
+  # Allows providing handy cross-reference to the build log
+  CIRRUS_TASK_ID: "{{env `CIRRUS_TASK_ID`}}"
+
+
+builders:
+  - type: amazon-ebs
+    name: win-server-wsl
+    source_ami_filter:
+      filters:
+        name: &win_release "Windows_Server-2022-English-Full-Base*"
+        root-device-type: ebs
+        virtualization-type: hvm
+      most_recent: true
+      owners:
+        - amazon
+    # While this image should run on metal, we can build it on smaller/cheaper systems 
+    instance_type: t3.large
+    force_deregister: true  # Remove AMI with same name if exists
+    force_delete_snapshot: true  # Also remove snapshots of force-removed AMI
+    # Note that we do not set shutdown_behavior to terminate, as a clean shutdown is required
+    # for windows provisioning to complete successfully.
+    communicator: winrm
+    winrm_username: Administrator # AWS provisions Administrator, unlike GCE 
+    winrm_insecure: true
+    winrm_use_ssl: true
+    winrm_timeout: 25m
+    # Script that runs on server start, needed to prep and enable winrm
+    user_data_file: '{{template_dir}}/bootstrap.ps1' 
+    # Required for network access, must be the 'default' group used by Cirrus-CI
+    security_group_id: "sg-042c75677872ef81c"
+    ami_name: &ami_name '{{build_name}}-w{{user `IMG_SFX`}}'
```

Comment:
When bringing updated images into CI (like podman) it's convenient to be able to just update the one `IMAGE_SUFFIX` variable and have it haul everything in together.  That's not going to be possible if the name suffix uses a `w` - at least not without lots of windows-specific changes to the CI setup.  I think maybe that can be avoided with either:

* Fudge it from a `w` to a 'c' here.
* Go whole-hog and merge `win_images/*` under `cache_images` (and into `cloud.yml`).

Either way, I see no reason to treat windows as a second-class citizen here.  It's fine to include it into the rest of the bunch.  I think that also means all the existing documentation will apply to the new windows images too, which would be nice.

WDYT?

Reply:
Happy either way on this. The yaml could be merged but we would need two CI tasks, unless we switch cache images to a gce instance as well. I ran into an issue with winrm communication from the container runner instance approach that cache images is currently using. I suspected it was firewall policy but did not debug it further.

Reply:
The container-based tasks have some odd quirks, I'm not surprised you found one.  Two tasks is fine by me, as is switching to VMs task for cache_images.  I dunno, I don't have super strong feelings beyond the `w` -> `c` change.  It's probably not going to do much for the diff size, but might be a bit easier to maintain if the templates and scripts are all in one place?  Hmmm.  We can let this marinate.

Reply:
Thought about it more.  Let's just leave the easy-thing in place (`w` -> `c`) for now.  If it makes sense later to merge all this content under `cache_images/` that can be done later.  My motivation is saving our effort for the "real" task here: getting this working in podman CI.